api/v1beta1/network_types.go

```diff
@@ -276,6 +276,24 @@ type SubnetSpec struct {
 
 	// Tags is a collection of tags describing the resource.
 	Tags Tags `json:"tags,omitempty"`
+
+	// ZoneType defines a zone type for this subnet.
+	//
+	// The valid values are availability-zone, local-zone, and wavelength-zone.
+	//
+	// Zone types local-zone or wavelength-zone is not selected to automatically create
+	// resources like: Nat Gateway, Network Load Balancers, control plane or compute nodes.
+	//
+	// When local-zone, the public subnets will be associated to the public route table,
+	// the private subnets uses the zone in the region to create route tables re-using
+	// Nat Gateways (preferred from parent zone, the zone type availability-zone in the region,
+	// or first available).
+	// +optional
+	ZoneType *string `json:"zoneType,omitempty"`
```

Comment:
This should probably be an enum and validated accordingly using kubebuilder tags